extra_proposal/proposal.py

```diff
@@ -0,0 +1,463 @@
+        for si in self._get_samples_mymdc():
+            _runs = []
+            for ri in runs_metadata:
+                if ri["sample_id"] == si["id"]:
+                    _runs.append(ri["run_number"])
+
+            name.append(si["name"])
+            index.append(si["id"])
+            url.append(si["url"])
+            description.append(si["description"])
+            runs.append(_runs)
+
+        return pd.DataFrame.from_dict(
+            {
+                "name": name,
+                "id": index,
+                "url": url,
+                "description": description,
+                "runs": runs,
+            }
+        )
+
+    @property
+    def instrument(self):
+        return self._proposal_directory.relative_to(DATA_ROOT_DIR).parts[0]
+
+    def info(self):
+        """Display information on a given proposal."""
+
+        # runs available in myMdC, and other information
+
+        def run_ranges(sequence):
+            # Adapted from:
+            #  https://stackoverflow.com/questions/3429510/pythonic-way-to-convert-a-list-of-integers-into-a-string-of-comma-separated-rang
+
+            # relevant to DAMNIT
+            sequence = np.unique(sequence)
+
+            grouped_sequence = (
+                list(x) for _, x in groupby(sequence, lambda x, c=count(): next(c) - x)
+            )
+            return (
+                ",".join(
+                    "-".join(map(str, (gi[0], gi[-1])[: len(gi)]))
+                    for gi in grouped_sequence
+                ),
+                sequence.size,
+            )
+
+        print(
+            "Proposal {} ── scientific instrument {}".format(
+                self._proposal_number, self.instrument
+            )
+        )
+        print(f"Data stored at {self._proposal_directory}")
+
+        # title
+        if self.title is not None:
+            print("'{}'".format(self.title))
+
+        # runs
+        runs = self.runs()
+        print("\nRuns collected: {} (total {})".format(*run_ranges(runs)))
+        if self.damnit is not None:
+            grouped_sequence, size = run_ranges(self.damnit.runs())
+            print(
+                " └── {:.1f}% processed by DAMNIT".format(100 * size / len(runs)),
+                end="",
+            )
+            if size != len(runs):
+                print(": {} (total {})".format(grouped_sequence, size))
+            else:
+                print(".")
+
+    def search_source(
+        self, regex: str, run: Optional[int | list[int]] = None
+    ) -> dict[list[str]]:
+        """Perform a case insensitive search of the regex in all data sources and aliases.
+
+        Args:
+            regex (str): The regular expression.
+            run (Optional[int  |  list[int]], optional): Specific run or list of runs. Defaults to None.
+
+        Returns:
+            list[list[str]]: A dictionary of runs, and for each a list of strings matching the regular expression.
+
+        Example:
+            proposal = Extra(1234)
```

Comment:
Nitpicking:
```suggestion
            proposal = Proposal(1234)
```

Reply:
I hadn't got to this method (or the mysterious `a` method below). I've just done a bit of clean-up now.